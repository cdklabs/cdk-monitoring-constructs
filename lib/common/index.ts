export * from "./alarm";
export * from "./metric";
export * from "./monitoring";
export * from "./url";
export * from "./widget";
export * from "./strings";
