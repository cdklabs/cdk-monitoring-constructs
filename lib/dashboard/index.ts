export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
