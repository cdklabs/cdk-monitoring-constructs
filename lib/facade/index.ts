export * from "./IMonitoringAspect";
export * from "./MonitoringFacade";
export * from "./MonitoringAspect";
