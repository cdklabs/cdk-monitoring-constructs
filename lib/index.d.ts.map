{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAC;AACzB,cAAc,aAAa,CAAC;AAC5B,cAAc,UAAU,CAAC;AACzB,cAAc,cAAc,CAAC"}