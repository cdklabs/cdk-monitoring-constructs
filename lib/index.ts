export * from "./common";
export * from "./dashboard";
export * from "./facade";
export * from "./monitoring";
