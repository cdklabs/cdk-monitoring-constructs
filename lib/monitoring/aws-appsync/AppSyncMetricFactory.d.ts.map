{"version":3,"file":"AppSyncMetricFactory.d.ts","sourceRoot":"","sources":["AppSyncMetricFactory.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,UAAU,EAAE,MAAM,4BAA4B,CAAC;AACxD,OAAO,EAAE,aAAa,EAAE,MAAM,4BAA4B,CAAC;AAE3D,OAAO,EACL,aAAa,EAEb,qBAAqB,EACtB,MAAM,cAAc,CAAC;AAItB,MAAM,WAAW,yBAAyB;IACxC;;OAEG;IACH,QAAQ,CAAC,GAAG,EAAE,UAAU,CAAC;IACzB;;;OAGG;IACH,QAAQ,CAAC,iBAAiB,CAAC,EAAE,OAAO,CAAC;IACrC;;;OAGG;IACH,QAAQ,CAAC,qBAAqB,CAAC,EAAE,qBAAqB,CAAC;CACxD;AAED,qBAAa,oBAAoB;IAC/B,SAAS,CAAC,QAAQ,CAAC,aAAa,EAAE,aAAa,CAAC;IAChD,SAAS,CAAC,QAAQ,CAAC,iBAAiB,EAAE,OAAO,CAAC;IAC9C,SAAS,CAAC,QAAQ,CAAC,qBAAqB,EAAE,qBAAqB,CAAC;IAChE,SAAS,CAAC,QAAQ,CAAC,aAAa,EAAE,aAAa,CAAC;gBAEpC,aAAa,EAAE,aAAa,EAAE,KAAK,EAAE,yBAAyB;IAU1E;;OAEG;IACH,SAAS;IAUT,iBAAiB;IAUjB,kBAAkB;IAWlB,wBAAwB;IAWxB,wBAAwB;IAWxB,wBAAwB;IAWxB,mBAAmB;IAWnB,kBAAkB;IAUlB,mBAAmB;IAWnB,kBAAkB;CASnB"}