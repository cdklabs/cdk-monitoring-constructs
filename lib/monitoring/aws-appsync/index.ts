export * from "./AppSyncMetricFactory";
export * from "./AppSyncMonitoring";
