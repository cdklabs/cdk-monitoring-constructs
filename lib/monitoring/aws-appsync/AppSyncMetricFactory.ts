import { IGraphqlApi } from "aws-cdk-lib/aws-appsync";
import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";

import {
  BaseMetricFactory,
  BaseMetricFactoryProps,
  MetricFactory,
  MetricStatistic,
  RateComputationMethod,
} from "../../common";

const Namespace = "AWS/AppSync";

export interface AppSyncMetricFactoryProps extends BaseMetricFactoryProps {
  /**
   * the GraphQL API to monitor
   */
  readonly api: IGraphqlApi;
  /**
   * whether the TPS should be filled with zeroes
   * @default - true
   */
  readonly fillTpsWithZeroes?: boolean;
  /**
   * method to compute TPS
   * @default - average
   */
  readonly rateComputationMethod?: RateComputationMethod;
}

export class AppSyncMetricFactory extends BaseMetricFactory<AppSyncMetricFactoryProps> {
  protected readonly fillTpsWithZeroes: boolean;
  protected readonly rateComputationMethod: RateComputationMethod;
  protected readonly dimensionsMap: DimensionsMap;

  constructor(metricFactory: MetricFactory, props: AppSyncMetricFactoryProps) {
    super(metricFactory, props);

    this.fillTpsWithZeroes = props.fillTpsWithZeroes ?? true;
    this.rateComputationMethod =
      props.rateComputationMethod ?? RateComputationMethod.AVERAGE;
    this.dimensionsMap = {
      GraphQLAPIId: props.api.apiId,
    };
  }

  /**
   * @deprecated use metricRequestRate
   */
  metricTps() {
    return this.metricFactory.toRate(
      this.metricRequestCount(),
      RateComputationMethod.PER_SECOND,
      true,
      "requests",
      this.fillTpsWithZeroes,
    );
  }

  metricRequestRate() {
    return this.metricFactory.toRate(
      this.metricRequestCount(),
      this.rateComputationMethod,
      true,
      "requests",
      this.fillTpsWithZeroes,
    );
  }

  metricRequestCount() {
    return this.metricFactory.createMetric(
      "Latency",
      MetricStatistic.N,
      "Requests",
      this.dimensionsMap,
      undefined,
      Namespace,
      undefined,
      this.region,
      this.account,
    );
  }

  metricLatencyP50InMillis() {
    return this.metricFactory.createMetric(
      "Latency",
      MetricStatistic.P50,
      "P50",
      this.dimensionsMap,
      undefined,
      Namespace,
      undefined,
      this.region,
      this.account,
    );
  }

  metricLatencyP90InMillis() {
    return this.metricFactory.createMetric(
      "Latency",
      MetricStatistic.P90,
      "P90",
      this.dimensionsMap,
      undefined,
      Namespace,
      undefined,
      this.region,
      this.account,
    );
  }

  metricLatencyP99InMillis() {
    return this.metricFactory.createMetric(
      "Latency",
      MetricStatistic.P99,
      "P99",
      this.dimensionsMap,
      undefined,
      Namespace,
      undefined,
      this.region,
      this.account,
    );
  }

  metric4XXErrorCount() {
    return this.metricFactory.createMetric(
      "4XXError",
      MetricStatistic.SUM,
      "4XX Error",
      this.dimensionsMap,
      undefined,
      Namespace,
      undefined,
      this.region,
      this.account,
    );
  }

  metric4XXErrorRate() {
    const metric = this.metric4XXErrorCount();
    return this.metricFactory.toRate(
      metric,
      this.rateComputationMethod,
      false,
      "errors",
    );
  }

  metric5XXFaultCount() {
    return this.metricFactory.createMetric(
      "5XXError",
      MetricStatistic.SUM,
      "5XX Fault",
      this.dimensionsMap,
      undefined,
      Namespace,
      undefined,
      this.region,
      this.account,
    );
  }

  metric5XXFaultRate() {
    const metric = this.metric5XXFaultCount();
    return this.metricFactory.toRate(
      metric,
      this.rateComputationMethod,
      false,
      "faults",
    );
  }
}
