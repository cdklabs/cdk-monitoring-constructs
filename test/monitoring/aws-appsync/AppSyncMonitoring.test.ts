import path from "path";
import { Duration, Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { GraphqlApi, IGraphqlApi, SchemaFile } from "aws-cdk-lib/aws-appsync";

import { AlarmWithAnnotation, AppSyncMonitoring } from "../../../lib";
import { addMonitoringDashboardsToStack } from "../../utils/SnapshotUtil";
import { TestMonitoringScope } from "../TestMonitoringScope";

test("snapshot test: no alarms", () => {
  const stack = new Stack();

  const scope = new TestMonitoringScope(stack, "Scope");

  const dummyApi = new GraphqlApi(stack, "testHttpApi", {
    name: "DummyApi",
    schema: SchemaFile.fromAsset(
      path.join(__dirname, "..", "..", "assets", "schema.graphql"),
    ),
  });

  const monitoring = new AppSyncMonitoring(scope, {
    api: dummyApi,
    humanReadableName: "Dummy API for testing",
    alarmFriendlyName: "DummyApi",
  });

  addMonitoringDashboardsToStack(stack, monitoring);
  expect(Template.fromStack(stack)).toMatchSnapshot();
});

test("snapshot test: no alarms with imported IGraphqlApi", () => {
  const stack = new Stack();

  const scope = new TestMonitoringScope(stack, "Scope");

  const dummyApi: IGraphqlApi = GraphqlApi.fromGraphqlApiAttributes(
    scope,
    "ImportedGraphQlApi",
    {
      graphqlApiId: "DummyApiFromElsewhere",
    },
  );

  const monitoring = new AppSyncMonitoring(scope, {
    api: dummyApi,
    humanReadableName: "Dummy API for testing",
    alarmFriendlyName: "DummyApi",
  });

  addMonitoringDashboardsToStack(stack, monitoring);
  expect(Template.fromStack(stack)).toMatchSnapshot();
});

test("snapshot test: all alarms", () => {
  const stack = new Stack();

  const scope = new TestMonitoringScope(stack, "Scope");

  let numAlarmsCreated = 0;

  const dummyApi = new GraphqlApi(stack, "testHttpApi", {
    name: "DummyApi",
    schema: SchemaFile.fromAsset(
      path.join(__dirname, "..", "..", "assets", "schema.graphql"),
    ),
  });

  const monitoring = new AppSyncMonitoring(scope, {
    api: dummyApi,
    humanReadableName: "Dummy API for testing",
    alarmFriendlyName: "DummyApi",
    addLatencyP50Alarm: {
      Warning: {
        maxLatency: Duration.millis(110),
      },
    },
    addLatencyP90Alarm: {
      Warning: {
        maxLatency: Duration.millis(220),
      },
    },
    addLatencyP99Alarm: {
      Warning: {
        maxLatency: Duration.millis(330),
      },
    },
    add4XXErrorCountAlarm: {
      Warning: {
        maxErrorCount: 0.02,
      },
    },
    add4XXErrorRateAlarm: {
      Warning: {
        maxErrorRate: 0.01,
      },
    },
    add5XXFaultCountAlarm: {
      Warning: {
        maxErrorCount: 2,
      },
    },
    add5XXFaultRateAlarm: {
      Warning: {
        maxErrorRate: 1,
      },
    },
    addLowTpsAlarm: {
      Warning: {
        minTps: 10,
      },
    },
    addHighTpsAlarm: {
      Warning: {
        maxTps: 100,
      },
    },
    useCreatedAlarms: {
      consume(alarms: AlarmWithAnnotation[]) {
        numAlarmsCreated = alarms.length;
      },
    },
  });

  addMonitoringDashboardsToStack(stack, monitoring);
  expect(numAlarmsCreated).toStrictEqual(9);
  expect(Template.fromStack(stack)).toMatchSnapshot();
});
