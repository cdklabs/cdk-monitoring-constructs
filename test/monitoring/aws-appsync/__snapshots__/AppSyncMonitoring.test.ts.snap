// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: all alarms 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ScopeTestDummyApiErrorCountWarning11BE36AF": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "Error count is too high.",
        "AlarmName": "Test-DummyApi-Error-Count-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Id": "m1",
            "Label": "4XX Error",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "4XXError",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "Sum",
            },
            "ReturnData": true,
          },
        ],
        "Threshold": 0.02,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiErrorRateWarningD181D1C6": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "Error rate is too high.",
        "AlarmName": "Test-DummyApi-Error-Rate-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Id": "m1",
            "Label": "4XX Error (avg)",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "4XXError",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "Average",
            },
            "ReturnData": true,
          },
        ],
        "Threshold": 0.01,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiFaultCountWarning6270BEC0": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "Fault count is too high.",
        "AlarmName": "Test-DummyApi-Fault-Count-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Id": "m1",
            "Label": "5XX Fault",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "5XXError",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "Sum",
            },
            "ReturnData": true,
          },
        ],
        "Threshold": 2,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiFaultRateWarning63CFB7F0": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "Fault rate is too high.",
        "AlarmName": "Test-DummyApi-Fault-Rate-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Id": "m1",
            "Label": "5XX Fault (avg)",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "5XXError",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "Average",
            },
            "ReturnData": true,
          },
        ],
        "Threshold": 1,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiLatencyP50Warning7EEB3E57": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "P50 latency is too high.",
        "AlarmName": "Test-DummyApi-Latency-P50-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Id": "m1",
            "Label": "P50",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "Latency",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "p50",
            },
            "ReturnData": true,
          },
        ],
        "Threshold": 110,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiLatencyP90Warning48639A66": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "P90 latency is too high.",
        "AlarmName": "Test-DummyApi-Latency-P90-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Id": "m1",
            "Label": "P90",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "Latency",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "p90",
            },
            "ReturnData": true,
          },
        ],
        "Threshold": 220,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiLatencyP99Warning254DA12C": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "P99 latency is too high.",
        "AlarmName": "Test-DummyApi-Latency-P99-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Id": "m1",
            "Label": "P99",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "Latency",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "p99",
            },
            "ReturnData": true,
          },
        ],
        "Threshold": 330,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiMaxTPSWarning62732F2B": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "TPS is too high.",
        "AlarmName": "Test-DummyApi-MaxTPS-Warning",
        "ComparisonOperator": "GreaterThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Expression": "FILL(requests,0) / PERIOD(requests)",
            "Id": "expr_1",
            "Label": "TPS",
          },
          Object {
            "Id": "requests",
            "Label": "Requests",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "Latency",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "SampleCount",
            },
            "ReturnData": false,
          },
        ],
        "Threshold": 100,
        "TreatMissingData": "missing",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "ScopeTestDummyApiMinTPSWarning4E8DB4D0": Object {
      "Properties": Object {
        "ActionsEnabled": true,
        "AlarmDescription": "TPS is too low.",
        "AlarmName": "Test-DummyApi-MinTPS-Warning",
        "ComparisonOperator": "LessThanThreshold",
        "DatapointsToAlarm": 3,
        "EvaluationPeriods": 3,
        "Metrics": Array [
          Object {
            "Expression": "FILL(requests,0) / PERIOD(requests)",
            "Id": "expr_1",
            "Label": "TPS",
          },
          Object {
            "Id": "requests",
            "Label": "Requests",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "GraphQLAPIId",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "testHttpApiC57B300F",
                        "ApiId",
                      ],
                    },
                  },
                ],
                "MetricName": "Latency",
                "Namespace": "AWS/AppSync",
              },
              "Period": 300,
              "Stat": "SampleCount",
            },
            "ReturnData": false,
          },
        ],
        "Threshold": 10,
        "TreatMissingData": "missing",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "testHttpApiC57B300F": Object {
      "Properties": Object {
        "AuthenticationType": "API_KEY",
        "Name": "DummyApi",
      },
      "Type": "AWS::AppSync::GraphQLApi",
    },
    "testHttpApiDefaultApiKey3EF156CA": Object {
      "DependsOn": Array [
        "testHttpApiSchemaDBC6E366",
      ],
      "Properties": Object {
        "ApiId": Object {
          "Fn::GetAtt": Array [
            "testHttpApiC57B300F",
            "ApiId",
          ],
        },
      },
      "Type": "AWS::AppSync::ApiKey",
    },
    "testHttpApiSchemaDBC6E366": Object {
      "Properties": Object {
        "ApiId": Object {
          "Fn::GetAtt": Array [
            "testHttpApiC57B300F",
            "ApiId",
          ],
        },
        "Definition": "",
      },
      "Type": "AWS::AppSync::GraphQLSchema",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: no alarms 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "testHttpApiC57B300F": Object {
      "Properties": Object {
        "AuthenticationType": "API_KEY",
        "Name": "DummyApi",
      },
      "Type": "AWS::AppSync::GraphQLApi",
    },
    "testHttpApiDefaultApiKey3EF156CA": Object {
      "DependsOn": Array [
        "testHttpApiSchemaDBC6E366",
      ],
      "Properties": Object {
        "ApiId": Object {
          "Fn::GetAtt": Array [
            "testHttpApiC57B300F",
            "ApiId",
          ],
        },
      },
      "Type": "AWS::AppSync::ApiKey",
    },
    "testHttpApiSchemaDBC6E366": Object {
      "Properties": Object {
        "ApiId": Object {
          "Fn::GetAtt": Array [
            "testHttpApiC57B300F",
            "ApiId",
          ],
        },
        "Definition": "",
      },
      "Type": "AWS::AppSync::GraphQLSchema",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
